src/Plugin/AbstractMethodPlugin.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace drupol\Yaroc\Plugin;
+
+use drupol\Yaroc\Http\AbstractClient;
+
+abstract class AbstractMethodPlugin extends AbstractClient implements MethodPluginInterface
+{
+    /**
+     * The endpoint.
+     *
+     * @var string
+     */
+    private $endpoint;
+
+    /**
+     * @var array
+     */
+    private $parameters = [];
+
+    /**
+     * {@inheritdoc}
+     */
+
+    /**
+     * @throws \Http\Client\Exception
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public function call()
+    {
+        $parameters = [
+            'jsonrpc' => '2.0',
+            'id'      => uniqid($this->getMethod() . '_'),
+            'params'  => $this->getParameters(),
+            'method'  => $this->getMethod(),
+        ];
+
+        $response = false;
+
+        try {
+            $response = $this->getHttpClient()->sendRequest(
+                $this->getMessageFactory()->createRequest(
+                    'POST',
+                    $this->getEndpoint(),
+                    [],
+                    json_encode(
+                        $parameters
+                    )
+                )
+            );
+        } catch (\Exception $exception) {
+            // @todo: log this.
+        }
+
+        return $response;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function withEndPoint(string $endpoint)
+    {
+        $clone = clone $this;
+
+        return $clone->setEndPoint($endpoint);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getEndPoint()
+    {
+        return $this->endpoint;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function withApiKey(string $apikey)
+    {
+        $clone = clone $this;
+
+        return $clone->setApiKey($apikey);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function withParameters(array $parameters)
+    {
+        $clone = clone $this;
+
+        return $clone->setParameters($parameters);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getParameters()
+    {
+        return $this->parameters;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMethod()
+    {
+        return $this::METHOD;
+    }
```

Comment:
This can be `static::METHOD`.
This also would allow to make the method itself static, but perhaps you prefer it to be non-static for other reasons (haven't studied the actual behavior of this class yet)

Reply:
Indeed, I don't know if I should make it static or not, yet.